src/wp-includes/post.php

```diff
@@ -3418,25 +3417,22 @@ function wp_count_posts( $type = 'post', $perm = '' ) {
 				FROM {$wpdb->posts}
 				WHERE post_type = %s AND post_status = 'private' AND post_author = %d
 			) AS filtered_posts
-			",
-			$type,
-			$type,
-			get_current_user_id()
-		);
+		";
+		$query_params = array( $type, $type, get_current_user_id() );
 	} else {
-		$query = $wpdb->prepare(
-			"
+		$query = "
 			SELECT post_status, COUNT(*) AS num_posts
 			FROM {$wpdb->posts}
 			WHERE post_type = %s
-			",
-			$type
-		);
+		";
+		$query_params = array( $type );
 	}
 
 	$query  .= ' GROUP BY post_status';
-	$results = (array) $wpdb->get_results( $query, ARRAY_A );
-	$counts  = array_fill_keys( get_post_stati(), 0 );
+
+	$prepared_query = $wpdb->prepare( $query, ...$query_params );
+	$results        = (array) $wpdb->get_results( $prepared_query, ARRAY_A );
```

Comment:
I'm combining these into one statement since both otherwise trigger the `WordPress.DB.PreparedSQL.NotPrepared` sniff.